### PR TITLE
add network config command

### PR DESCRIPTION
**Pull request by wlan0**
@ibuildthecloud

### CODE REVIEW

**Comment by sidharthamani** [comment]
added support for gateway and MTU

File: cmd/network/network.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+
+	//post run
+	networkConf := cfg.Network
+	if isSet(networkConf) {
+		if isSet(cfg.Network.PostRun) {
```
Comment (sidharthamani) [inline cmd/network/network.go]: switch to pointer

File: cmd/network/network.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			container := docker.NewContainer(config.DOCKER_HOST, &(cfg.Network.PostRun))
+			container.StartAndWait()
```
Comment (sidharthamani) [inline cmd/network/network.go]: at least log error

File: cmd/network/network.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		if int(link[i]) != int(c) && c != '*' {
+			return false
+		}	
+	}
+	return true	
+}
+
+func getNetConfs(cfg *config.Config) []config.NetConf {
+	networkConf := cfg.Network
+	if &networkConf != nil {
```
Comment (sidharthamani) [inline cmd/network/network.go]: cleanup

File: cmd/network/network.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+		if networkConf.Interfaces != nil {
```
Comment (sidharthamani) [inline cmd/network/network.go]: never nil

File: config/config.go [hunk 5]
```diff
@@ -55,6 +55,20 @@ type Config struct {
...
 	Addons           map[string]Config `yaml:"addons,omitempty"`
+	Network    	 NetworkConfig	   `yaml:"network_config,omitempty"`
+}
+
+type NetworkConfig struct {
+	Interfaces	[]NetConf 	`yaml:"config"`
+	PostRun 	ContainerConfig	`yaml:"post_run"`
+}
+
+type NetConf struct {
```
Comment (sidharthamani) [inline config/config.go]: rename InterfaceConfig

File: cmd/network/network.go [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+			container.StartAndWait()
+		}
+	}
+	return nil
+}
+
+func applyNetConf(link netlink.Link, netConf config.NetConf) error {
+	if matches(link.Attrs().Name, netConf.Match) {
+		if netConf.DHCP {
+			cmd := exec.Command("udhcpc", "-i", link.Attrs().Name)
```
Comment (sidharthamani) [inline cmd/network/network.go]: set cmd.Stdout = os.Stdout and same with err

File: cmd/network/network.go [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+			cmd.Run()
```
Comment (sidharthamani) [inline cmd/network/network.go]: check result and log error

File: cmd/network/network.go [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+			}
+		}
+		if netConf.Gateway != "" {
+			route := netlink.Route{LinkIndex: link.Attrs().Index, Scope: netlink.SCOPE_LINK, Gw: net.ParseIP(netConf.Gateway)}
+			if err := netlink.RouteAdd(&route); err != nil {
+				log.Error("gateway set failed")
+				return err
+			}
+		}
+		if err := netlink.LinkSetUp(link); err != nil {
```
Comment (sidharthamani) [inline cmd/network/network.go]: what if link is up

Comment (wlan0) [inline cmd/network/network.go]: if link is up, it doesn't throw an error

**Pull request merged**